task_02/roman_kuprii/main.cpp

```diff
@@ -0,0 +1,99 @@
+#include <iostream>
+#include <vector>
+#include <fstream>
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+//using namespace std;
+
+struct Vertex {
+    int id;
+    int edges;
+
+    Vertex(): id(-1), edges(0) {}
+    Vertex(int _id): id(_id), edges(0) {}
+    Vertex(int _id, int edg): id(_id), edges(edg) {}
+};
+
+struct Edge {
+    int id;
+    int node[2];
+
+    Edge(int start, int end, int id) {
+        node[0] = start;
+        node[1] = end;
+        this->id = id;
+    }
+};
+
+class Graph {
+//privat
+int size;
+vector<Vertex> vertices;
+vector<Edge> edges;
+
+public:
+    Graph(): size(0) {}
+
+    Graph(const vector<Edge> & init_edges, const vector<Vertex> & init_vertices) {
+        vertices = init_vertices;
+        edges = init_edges;
+        size = vertices.size();
+    }
+
+    ~Graph() {
+        vertices.clear();
+        edges.clear();
+    }
+
+    int write_json() {
+        std::ofstream out;
+        out.open("graph.json", std::ofstream::out | std::ofstream::trunc);
+        out << "{\n\t\"vertices\": [\n\t\t";
+        for (int i = 0; i < size; i++) {
+            out << "{\n\t\t\t\"id\": " << i << ",\n\t\t\t\"edge_ids\": [";
+            int cnt(0);
+            for (vector<Edge>::iterator it = edges.begin(); it != edges.end(); it++) {
+                if (((*it).node[0] == i) || ((*it).node[1] == i)) {
+                    cnt++;
+                    if (cnt < vertices[i].edges) {
+                        out << it->id << ", ";
+                    } else out << (*it).id;
+                }
+            }
+            out << "]\n\t\t}";
+            if (i != (size-1)) out << ", ";
+        }
+        out << "\n\t],\n\t\"edges\": [\n\t\t";
+        for(auto it = edges.begin(); it != edges.end(); it++) {
+            out << "{\n\t\t\t\"id\": " << it->id << ",\n\t\t\t\"vertex_ids\": [" << it->node[0] << ", " << it->node[1] << "]\n\t\t}";
+            if (it != edges.end()-1) {
+                out << ", ";
+            }
+        }
+        out << "\n\t]\n}";
+        out.close();
+        return 0;
+    }
+
+};
+
+int main(int argc, char**argv) {
```

Comment:
Зачем тебе аргументы окружения?
```suggestion
int main() {
```